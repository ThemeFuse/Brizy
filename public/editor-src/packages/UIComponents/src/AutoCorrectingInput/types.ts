import { Ref } from "react";
import { AutoCorrectingInput } from "./index";

export interface CommonProps {
  className?: string;
  value: number;
  onChange: (value: number) => void;
  min: number;
  max: number;
  step: number;
  onFocus?: VoidFunction;
  onBlur?: VoidFunction;
  onTextChange?: (e: string | number) => void;
  onMouseEnter?: VoidFunction;
  onMouseLeave?: VoidFunction;
  size?: number;
  handleDecrementCb?: () => void;
}

export interface StepperProps extends CommonProps {
  stepperSize: "small" | "medium" | "large" | "auto";
  handleDecrementCb: () => void;
  inputRef: Ref<AutoCorrectingInput>;
}

export interface State {
  text: string | number;
  value: number;
  prevPropsValue: number;
}
