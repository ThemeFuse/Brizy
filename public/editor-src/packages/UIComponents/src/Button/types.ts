import { ReactNode } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  onClick?: VoidFunction;
  reverse?: boolean;
  children?: ReactNode;
  className?: string;
  icon?: IconsName;
  align?: "left" | "center" | "right";
  reverseTheme?: boolean;
  title?: string;
}
