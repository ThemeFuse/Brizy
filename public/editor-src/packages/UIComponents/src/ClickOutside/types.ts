import { ReactElement, RefObject } from "react";

type FunctionalException = (el: HTMLElement) => boolean;

export type Exception = string | FunctionalException;

export type Acc = {
  functionExceptions: FunctionalException[];
  stringExceptions: string[];
};

export interface Props<T extends HTMLElement> {
  children: (ref: RefObject<T>) => ReactElement;
  exceptions: Exception[];
  onClickOutside: (e: MouseEvent) => void;
}
