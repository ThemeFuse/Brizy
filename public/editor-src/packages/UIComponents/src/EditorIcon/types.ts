export enum IconsName {
  Close = "Close",
  CheckSmall = "CheckSmall",
  Spinner = "Spinner",
  PlusCircle = "PlusCircle",
  Duplicate = "Duplicate",
  CheckCircle = "CheckCircle",
  Settings = "Settings",
  Heart = "Heart",
  StreUp = "StreUp",
  StreDown = "StreDown",
  TextAlignCenter = "TextAlignCenter",
  TextAlignLeft = "TextAlignLeft",
  TextAlignRight = "TextAlignRight",
  Circle2 = "Circle2",
  CircleRemove = "CircleRemove",
  "t2-sermon-list" = "MinistryBrandsSermonList",
  "t2-sermon-detail" = "MinistryBrandsSermonDetail",
  "t2-sermon-featured" = "MinistryBrandsSermonFeatured",
  "t2-sermon-layout" = "MinistryBrandsSermonLayout",
  "t2-event-list" = "MinistryBrandsEventList",
  "t2-event-detail" = "MinistryBrandsEventDetail",
  "t2-event-featured" = "MinistryBrandsEventFeatured",
  "t2-event-layout" = "MinistryBrandsEventLayout",
  "t2-event-calendar" = "MinistryBrandsEventCalendar",
  "t2-group-list" = "MinistryBrandsGroupList",
  "t2-group-detail" = "MinistryBrandsGroupDetail",
  "t2-group-featured" = "MinistryBrandsGroupFeatured",
  "t2-group-slider" = "MinistryBrandsGroupSlider",
  "t2-group-layout" = "MinistryBrandsGroupLayout",
  "nc-stre-up" = "StreUp",
  "nc-stre-down" = "StreDown",
  MediaImage = "MediaImage",
  MediaVideo = "MediaVideo",
  MediaMap = "MediaMap",
  Reverse = "Reverse"
}
