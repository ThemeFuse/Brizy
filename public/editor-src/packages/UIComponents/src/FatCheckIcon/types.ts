import { MouseEventHandler } from "react";
import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  onCheck?: VoidFunction;
  isActive?: boolean;
  isChecked?: boolean;
}

export interface CheckMarkProps extends WithClassName {
  isChecked: boolean;
  onClick?: MouseEventHandler;
}
