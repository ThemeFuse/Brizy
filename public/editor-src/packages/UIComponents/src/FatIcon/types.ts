import { IconsName } from "../EditorIcon/types";
import { WithClassName } from "../types/attributes";

export interface Props extends WithClassName {
  icon: IconsName;
  label: string;
  onClick?: VoidFunction;
  isActive?: boolean;
}
