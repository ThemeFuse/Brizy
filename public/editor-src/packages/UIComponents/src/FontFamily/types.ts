import { FontFamilyType } from "../types/fonts/familyType";
import { Weight } from "../types/fonts/weight";

export type FontsBlock = Partial<{
  config: Array<FontObject>;
  blocks: Array<FontObject>;
  [FontFamilyType.google]: Array<FontObject>;
  [FontFamilyType.upload]: Array<FontObject>;
}>;

export interface Props {
  addFont?: VoidFunction;
  onChange: (font: FontWithType) => void;
  value: string;
  className?: string;
  addFontLabel?: string;
  fonts: FontsBlock;
}
export interface FontSizes {
  default: string;
  great_vibes: string;
  alex_brush: string;
  allura: string;
  parisienne: string;
}

export interface FontObject {
  brizyId?: string;
  family: string;
  id: string;
  title: string;
  weights: Array<Weight>;
}

export interface FontWithType {
  id: string;
  type: FontFamilyType;
  weights: Array<Weight>;
}

export interface FontList {
  fonts: Array<FontObject>;
  type: FontFamilyType;
}
