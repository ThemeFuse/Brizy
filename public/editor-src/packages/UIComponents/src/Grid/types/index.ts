import { ReactElement } from "react";

export interface ColumnProps {
  className?: string;
  align: "start" | "center" | "end";
}

export interface GridProps {
  className?: string;
  grid: Array<"auto" | number>;
  showSeparator?: boolean;
  children: ReactElement<ColumnProps>[];
}
