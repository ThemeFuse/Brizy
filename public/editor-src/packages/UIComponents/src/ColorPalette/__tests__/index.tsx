import { fireEvent, render, waitFor } from "@testing-library/react";
import React from "react";
import { Hex } from "../../types/Hex";
import { ColorPalette } from "../index";
import { PaletteObject } from "../types";

const mockPalette: PaletteObject<string>[] = [
  { id: "red", hex: "#FF0000" as Hex },
  { id: "green", hex: "#00FF00" as Hex },
  { id: "blue", hex: "#0000FF" as Hex }
];

const onChange = jest.fn();
const onOpenSettings = jest.fn();
describe("Test ColorPalette", () => {
  describe("snapshots", () => {
    test("Default usage", async () => {
      const palette = render(
        <ColorPalette
          palette={mockPalette}
          onChange={onChange}
          value={"red"}
          openSettings={onOpenSettings}
        />
      );

      await waitFor(() => {
        expect(palette.container.firstChild).toBeInTheDocument();
      });

      expect(palette.container.firstChild).toMatchSnapshot();
    });
  });

  test("Renders div with correct className", () => {
    const { container } = render(
      <ColorPalette onChange={onChange} value={"red"} palette={mockPalette} />
    );
    expect(container.firstChild).toHaveClass("brz-ed-control__color-palette");
  });

  test("Renders div with correct className and custom className", () => {
    const { container } = render(
      <ColorPalette
        className="custom-class"
        palette={mockPalette}
        onChange={onChange}
        value={"red"}
      />
    );
    expect(container.firstChild).toHaveClass("brz-ed-control__color-palette");
    expect(container.firstChild).toHaveClass("custom-class");
  });

  test("Renders div with the correct number of color items", () => {
    const { container } = render(
      <ColorPalette palette={mockPalette} onChange={onChange} value={"red"} />
    );
    expect(
      container.querySelectorAll(".brz-ed-control__color-palette__item")
    ).toHaveLength(3);
  });

  test("Check onChange", () => {
    const { container } = render(
      <ColorPalette palette={mockPalette} value="red" onChange={onChange} />
    );

    const node = container.querySelectorAll(
      ".brz-ed-control__color-palette__item"
    )[1];

    fireEvent.click(node);
    expect(onChange).toHaveBeenCalledWith("green");
  });

  test("Render icon and call openSettings", async () => {
    const { container } = render(
      <ColorPalette
        palette={mockPalette}
        openSettings={onOpenSettings}
        value={""}
        onChange={onChange}
      />
    );

    const settingsIcon = container.querySelector(
      ".brz-ed-control__color-palette__icon"
    );

    await waitFor(() => {
      expect(settingsIcon).toBeInTheDocument();
    });

    settingsIcon && fireEvent.click(settingsIcon);
    expect(onOpenSettings).toHaveBeenCalled();
  });

  test("Do not render icon when no openSettings prop", async () => {
    const { container } = render(
      <ColorPalette palette={mockPalette} value={""} onChange={onChange} />
    );

    expect(
      container.querySelector(".brz-ed-control__color-palette__icon")
    ).toBeNull();
  });

  test("Render active className", () => {
    const { container } = render(
      <ColorPalette palette={mockPalette} value="green" onChange={onChange} />
    );
    expect(
      container.querySelectorAll(".brz-ed-control__color-palette__item")[1]
    ).toHaveClass("active");
  });
});
