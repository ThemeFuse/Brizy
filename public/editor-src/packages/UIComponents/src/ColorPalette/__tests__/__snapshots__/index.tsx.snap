// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ColorPalette snapshots Default usage 1`] = `
<div
  class="brz-ed-control__color-palette"
>
  <div
    class="brz-ed-control__color-palette__item active"
    style="background-color: rgb(255, 0, 0);"
  />
  <div
    class="brz-ed-control__color-palette__item"
    style="background-color: rgb(0, 255, 0);"
  />
  <div
    class="brz-ed-control__color-palette__item"
    style="background-color: rgb(0, 0, 255);"
  />
  <div
    class="brz-ed-control__color-palette__icon"
  >
    <svg
      class="brz-icon-svg brz-ed-icon-svg align-[initial]"
      fill="currentColor"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <g
        fill="currentColor"
        stroke="none"
      >
        <g
          fill="currentColor"
          transform="translate(-273.000000, -132.000000)"
        >
          <g
            transform="translate(47.000000, 0.000000)"
          >
            <g
              transform="translate(0.000000, 78.000000)"
            >
              <g
                transform="translate(0.000000, 40.000000)"
              >
                <g
                  transform="translate(229.000000, 14.000000)"
                >
                  <path
                    d="M10.3,5.2 L11.4,3.1 L10,1.7 L7.9,2.8 C7.6,2.6 7.2,2.5 6.8,2.4 L6,0 L4,0 L3.2,2.3 C2.9,2.4 2.5,2.5 2.2,2.7 L0.1,1.6 L-1.4,3.1 L-0.3,5.2 C-0.5,5.5 -0.6,5.9 -0.7,6.2 L-3,7 L-3,9 L-0.7,9.8 C-0.6,10.2 -0.4,10.5 -0.3,10.9 L-1.4,13 L0,14.4 L2.1,13.3 C2.4,13.5 2.8,13.6 3.2,13.7 L4,16 L6,16 L6.8,13.7 C7.2,13.6 7.5,13.4 7.9,13.3 L10,14.4 L11.4,13 L10.3,10.9 C10.5,10.6 10.6,10.2 10.7,9.8 L13,9 L13,7 L10.7,6.2 C10.6,5.9 10.5,5.5 10.3,5.2 Z M5,11 C3.3,11 2,9.7 2,8 C2,6.3 3.3,5 5,5 C6.7,5 8,6.3 8,8 C8,9.7 6.7,11 5,11 Z"
                  />
                </g>
              </g>
            </g>
          </g>
        </g>
      </g>
    </svg>
  </div>
</div>
`;
